include/circt/Dialect/Moore/MooreTypes.td

```diff
@@ -305,6 +305,19 @@ def UnpackedStructType : StructLikeType<
   }];
 }
 
+def RefType : MooreTypeDef<"Ref", []>{
+  let mnemonic = "ref";
+  let parameters = (ins "::mlir::Type":$nestedType);
```

Comment:
You could probably make the nested type an `UnpackedType`, to accept more or less any SV type here. That would be more specific than MLIR `Type`, which could be any weird thing in MLIR that we don't know how to work with.

Reply:
You're totally right! :smile: